### ISSUE TITLE
Omitting slash mates (/1 or /2) in read names before QNAME truncating

### ISSUE THREAD

**Issue by ronin-gw** [BenLangmead/bowtie2#265]
In [`printReadName`](https://github.com/BenLangmead/bowtie2/blob/eff7f6e0265e805f5e10fd6dedf4266bdb5ce64d/sam.h#L216-L240), read names are truncated by following order:

1. Omit slash mate (`/1` or `/2`) from the end of a read name
2. Keep first 255 characters
3. Ignore characters after the first whitespace

However, read names in some fastq files (maybe following the old illumina format) can have both slash mates and additional strings. For example, these are the heads of the paired-end fastq files I got from the [ENCODE](https://www.encodeproject.org/experiments/ENCSR975GZA/).
```
@BI:SL-HEA:C3BFGACXX:2:2316:10503:45861/1 1:X:0:CATAGCGA
TAGGGTTAGGGTTAGGGTTAGGGTT
+
@@@FFADDFHHCFHJJJCGDHIJGG
```
```
@BI:SL-HEA:C3BFGACXX:2:2316:10503:45861/2 1:X:0:CATAGCGA
TAACCCTAACCCTAACCCTAACCCT
+
CCCFFFFFHHHHHJJIJIIJJJJII
```

So I think "slash mates" should be omitted *after* the other truncations to match read names as pairs.

**Comment by ch4rr0** [comment]
Good catch. I'll have this fix committed soon and will include this change in the next release.

**Comment by ch4rr0** [comment]
I think this recent commit should address this issue. Here's an example output using the example reads you provided:

```
BI:SL-HEA:C3BFGACXX:2:2316:10503:45861  77      *       0       0       *       *       0       0       TAGGGTTAGGGTTAGGGTTAGGGTT       @@@FFADDFHHCFHJJJCGDHIJGG       YT:Z:UP
BI:SL-HEA:C3BFGACXX:2:2316:10503:45861  141     *       0       0       *       *       0       0       TAACCCTAACCCTAACCCTAACCCT       CCCFFFFFHHHHHJJIJIIJJJJII       YT:Z:UP
```

**Comment by ch4rr0** [comment]
This change is available in v2.4.0 and later.
